tests/unittests/tests-analog_util/tests-analog_util.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2018 Eistec AB
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include "embUnit.h"
+#include "tests-analog_util.h"
+
+#include "analog_util.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+typedef struct {
+    int expected;
+    int sample;
+    int min;
+    int max;
+    adc_res_t res;
+} test_values_t;
+
+/* Arbitrarily chosen test vectors */
+/* TODO: Choose test vectors in a more qualified manner to catch any edge cases */
```

Comment:
Compared to the previous version, its dividing by a resolution value which is one bigger, would be good I think to show a edge case like this.

Reply:
Also test does not pass on `samr21-xpro` with the previous version on commit `8edcbd1... unittests: Add analog_util tests`, so would be nice to have a working test fand an updated test when changing the function.

Reply:
I don't see the point in rewriting the test to fit the broken unused version in current master, it seems like a lot of work just for immediately dumping something to the history archive.

The broken implementation in master does not even link for the STM32 because the scaling factor array becomes several hundred megabytes large because of the enum values used as indices.

Regarding the division, I should have mentioned it in the PR description: Since the ADC measures a continuous range of physical values, each quantized ADC output value corresponds to a continuous range of input values. Since the ADC has `2^N` counts, each output corresponds to a range of `range/2^N` values (LSB).
Also, it is much more efficient to divide by `2^N` than `(2^N)-1`
Some relevant sources:
 - Kinetis K22F reference manual, ADC chapter
  >The ADC quantizes the ideal straight-line transfer function into 65536 steps (in the 16-bit mode). Each step ideally has the same height, that is, 1 code, and width. The width is defined as the delta between the transition points to one code and the next. The ideal code width for an N-bit converter, where N can be 16, 12, 10, or 8, defined as 1 LSB, is:
  >![1LSB=(Vrefh)/2^N](https://user-images.githubusercontent.com/317506/36427947-db0ed952-164e-11e8-9eb8-86a26b37ca86.png)

 - https://electronics.stackexchange.com/a/258497 (mentions that the same formula applies for Atmega128)

Reply:
"Was broken and unused before" is a good justification for not doing additional work on this.